cmake/Modules/FindMKL.cmake

```diff
@@ -116,30 +120,52 @@ MACRO(CHECK_ALL_LIBRARIES LIBRARIES _name _list _flags)
   # start checking
   SET(_libraries_work TRUE)
   SET(${LIBRARIES})
+  SET(${OPENMP_TYPE})
+  SET(${OPENMP_LIBRARY})
   SET(_combined_name)
+  SET(_openmp_type)
+  SET(_openmp_library)
   SET(_paths)
-  set(__list)
-  foreach(_elem ${_list})
-    if(__list)
-      set(__list "${__list} - ${_elem}")
-    else(__list)
-      set(__list "${_elem}")
-    endif(__list)
-  endforeach(_elem)
   IF (NOT MKL_FIND_QUIETLY)
-    message(STATUS "Checking for [${__list}]")
+    set(_str_list)
+    foreach(_elem ${_list})
+      if(_str_list)
+        set(_str_list "${_str_list} - ${_elem}")
+      else()
+        set(_str_list "${_elem}")
+      endif()
+    endforeach(_elem)
+    message(STATUS "Checking for [${_str_list}]")
   ENDIF ()
   FOREACH(_library ${_list})
     SET(_combined_name ${_combined_name}_${_library})
+    UNSET(${_prefix}_${_library}_LIBRARY)
     IF(_libraries_work)
-      IF(${_library} STREQUAL "gomp")
-          FIND_PACKAGE(OpenMP)
+      IF(${_library} MATCHES "omp")
+        IF(_openmp_type)
+          MESSAGE(FATAL_ERROR "More than one OpenMP libraries appear in the MKL test: ${_list}")
+        ELSEIF(${_library} MATCHES "gomp")
+          SET(_openmp_type "GNU")
+          # Use FindOpenMP to find gomp
+          FIND_PACKAGE(OpenMP QUIET)
```

Comment:
oh... findmkl and findomp probably should not recursively call into each other :P